src/libecalc/common/graph.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from typing import Dict, Generic, List, Protocol, TypeVar
+
+import networkx as nx
+
+NodeID = str
+
+
+class NodeWithID(Protocol):
```

Comment:
should probably make sure that the `id` is immutable/frozen, or even the node? but that the graph may change?